tests/cloudpickle_test.py

```diff
@@ -2130,6 +2137,38 @@ def test_pickle_importable_typevar(self):
         from typing import AnyStr
         assert AnyStr is pickle_depickle(AnyStr, protocol=self.protocol)
 
+    @unittest.skipIf(sys.version_info < (3, 7),
+                     "Pickling generics not supported below py37")
+    def test_generic_type(self):
+        for obj in _generic_objects_to_test():
+            pickle_depickle(obj, protocol=self.protocol)
+
+    @unittest.skipIf(sys.version_info < (3, 7),
+                     "Pickling type hints not supported below py37")
+    def test_locally_defined_class_with_type_hints(self):
+        with subprocess_worker(protocol=self.protocol) as worker:
+            for type_ in _generic_objects_to_test():
+                # The type annotation syntax causes a SyntaxError on Python 3.5
+                code = textwrap.dedent("""\
+                class MyClass:
+                    attribute: type_
+
+                    def method(self, arg: type_) -> type_:
+                        return arg
+                """)
+                ns = {"type_": type_}
+                exec(code, ns)
+                MyClass = ns["MyClass"]
+
+                def check_annotations(obj, expected_type):
+                    assert obj.__annotations__["attribute"] is expected_type
+                    assert obj.method.__annotations__["arg"] is expected_type
```

Comment:
I forgot to test the return type. Let's do it for consistency:

```suggestion
                    assert obj.__annotations__["attribute"] is expected_type
                    assert obj.method.__annotations__["arg"] is expected_type
                    assert obj.method.__annotations__["return"] is expected_type
```

Reply:
Would we want to test that `type_` correctly subclasses `typing.Generic[T]`, we would need something like:
```python
 assert expected_type in obj.__annotations__['attribute'].__orig_bases__
```
with `expected_type` being the `Generic` instance the   `C` class subclasses from.

Reply:
Ok but this should rather be done in a dedicated atomic test rather than in the generic integration test for dynamic class annotations.

Reply:
Let me commit this a add a new test.